src/cljs/athens/listeners.cljs

```diff
@@ -82,45 +83,55 @@
+                                       KeyCodes.DASH      (dispatch [:zoom/out])
+                                       KeyCodes.ZERO      (dispatch [:zoom/reset])
+                                       KeyCodes.K         (dispatch [:athena/toggle])
+                                       KeyCodes.G         (dispatch [:devtool/toggle])
+                                       KeyCodes.Z         (let [editing-uid    @(subscribe [:editing/uid])
+                                                                selected-items @(subscribe [::select-subs/items])]
+                                                            ;; editing/uid must be nil or selected-items must be non-empty
+                                                            (when (or (nil? editing-uid)
+                                                                      (not-empty selected-items))
+                                                              (if shift
+                                                                (dispatch [:redo])
+                                                                (dispatch [:undo]))))
                                        KeyCodes.BACKSLASH (if shift
                                                             (dispatch [:right-sidebar/toggle])
                                                             (dispatch [:left-sidebar/toggle]))
-
-                                       KeyCodes.COMMA (router/navigate :settings)
-
-                                       KeyCodes.T (util/toggle-10x)
+                                       KeyCodes.COMMA     (router/navigate :settings)
+                                       KeyCodes.T         (util/toggle-10x)
                                        nil)
-      alt (condp = key-code
-            KeyCodes.LEFT (when (nil? editing-uid) (.back js/window.history))
-            KeyCodes.RIGHT (when (nil? editing-uid) (.forward js/window.history))
-            KeyCodes.D (router/nav-daily-notes)
-            KeyCodes.G (router/navigate :graph)
-            KeyCodes.A (router/navigate :pages)
-            KeyCodes.T (dispatch [:theme/toggle])
-            nil))))
+      alt                           (condp = key-code
+                                      KeyCodes.D     (router/nav-daily-notes)
+                                      KeyCodes.G     (router/navigate :graph)
+                                      KeyCodes.A     (router/navigate :pages)
+                                      KeyCodes.T     (dispatch [:theme/toggle])
+                                      nil)
+      (util/navigate-key? meta alt) (condp = key-code
+                                      KeyCodes.LEFT  (when (or (nil? editing-uid)
+                                                               start?)
+                                                       (.back js/window.history))
+                                      KeyCodes.RIGHT (when (or (nil? editing-uid)
```

Comment:
Why `(nil? editing-uid)` here and also `if` statements for `editing-uid` on lines 92 and 95? Seems redundant.